framework/src/media/Decoder.cpp

```diff
@@ -89,22 +117,55 @@ bool Decoder::getFrame(unsigned char *buf, size_t *size, unsigned int *sampleRat
 		return false;
 	}
 
-	if (_get_frame(&mPlayer)) {
-		pcm_data_t pcm;
-		if (_frame_decoder(&mPlayer, &pcm) != PV_FAILURE) {
-			*size = pcm.length * sizeof(short);
-			*sampleRate = pcm.samplerate;
-			*channels = pcm.channels;
-			memcpy(buf, pcm.samples, *size);
-			return true;
-		} else {
-			meddbg("Error: Decoding frame failed!\n");
-			return false;
+	while (1) {
+		if (pcm.samples) {
```

Comment:
I think you can provide function to handle line #121~ 145.
Also pv_player_get_frame & pv_player_frame_decode can be merged because we don't process anything after get_frame alone.
Of course you can do that after this merged, but let me hear your answer.

Reply:
Yes, it's more reasonable.
I will move this logic to pv_player and provide new function.

Reply:
Fixed